src/repo.rs

```diff
@@ -2043,6 +2043,20 @@ impl Repository {
     }
 
     /// Test if the ignore rules apply to a given path.
+    #[cfg(windows)]
```

Comment:
Instead of `#[cfg]`, could `if cfg!(windows)` be used to conditionally do the `to_string_lossy` instead?